packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -266,33 +266,35 @@ export const vModelDynamic: ObjectDirective<
   }
 }
 
-function callModelHook(
-  el: HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement,
-  binding: DirectiveBinding,
-  vnode: VNode,
-  prevVNode: VNode | null,
-  hook: keyof ObjectDirective
-) {
-  let modelToUse: ObjectDirective
-  switch (el.tagName) {
+function resolveDynamicModel(tagName: string, type: string | undefined) {
+  switch (tagName.toUpperCase()) {
```

Comment:
I think the conversion to uppercase is unneeded here.
https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName

Reply:
For `callModelHook` (which passes in `el.tagName`), that's true. But `vModelDynamic.getSSRProps` passes in `vnode.type`, which is lowercase, that's why this is here. That's unclear though, so I'll add a comment.